### PR TITLE
Refactor 1

### PR DESCRIPTION
**Pull request by jjaworska**

### CODE REVIEW

File: src/main/kotlin/cacophony/automata/DFAEquivalenceHelper.kt [hunk 1]
```diff
@@ -69,15 +65,13 @@ private fun <StateA, StateB, AtomType, ResultType> initializeDistinguishableStat
...
-    return productions.keys.map { it.second }.toSet()
-}
+private fun <State, AtomT> getSymbols(productions: Map<Pair<State, AtomT>, State>): Set<AtomT> = productions.keys.map { it.second }.toSet()
 
-private fun <State, AtomType> invertProductions(
-    dfa: DFA<State, AtomType, *>,
-    symbols: Set<AtomType>,
-): Map<Pair<State?, AtomType>, Set<State?>> {
-    val inverted = mutableMapOf<Pair<State?, AtomType>, MutableSet<State?>>()
+private fun <State, AtomT> invertProductions(
```
Comment (Rafi0221) [inline src/main/kotlin/cacophony/automata/DFAEquivalenceHelper.kt:70]: Why not `StateT`?

**Review by Rafi0221 — APPROVED** [review]
OK, there is one small thing you might change

File: src/main/kotlin/cacophony/automata/DFAEquivalenceHelper.kt (hunk 1 above)
Comment (jjaworska) [reply src/main/kotlin/cacophony/automata/DFAEquivalenceHelper.kt]: Yeah, thanks, actually I spoted a few more places, take a look

Comment (Rafi0221) [reply src/main/kotlin/cacophony/automata/DFAEquivalenceHelper.kt]: Fine by me. You can merge

**Pull request merged**